src/GlyphCache.cpp

```diff
@@ -357,8 +358,9 @@ const GlyphFace * GlyphCache::Loader::read_glyph(unsigned short glyphid, GlyphFa
         if (_glyf)
         {
             int xMin, yMin, xMax, yMax;
-            size_t locidx = TtfUtil::LocaLookup(glyphid, _loca, _loca.size(), _head);
-            void *pGlyph = TtfUtil::GlyfLookup(_glyf, locidx, _glyf.size());
+            int locidx = TtfUtil::LocaLookup(glyphid, _loca, _loca.size(), _head);
+            assert(locidx >= 0);
+            void *pGlyph = TtfUtil::GlyfLookup(_glyf, static_cast<size_t>(locidx), _glyf.size());
```

Comment:
Depending on the necessity of having a value filled into that bbox further down in this method, I suggest something like
Glyph *pGlyph = nullptr; /* is there such a generic type to avoid all the casting? */
int locidx = ...LocaLookup...;
if (locidx >= 0) pGlyph = ...GlyfLookup...;
if (pGlyph && ...GlyfBox...) ...etc...

Reply:
LocaLookup returns -1 and -2 as sentinel error values which are tested for in GlyfLookup. The assert() would cause a crash on faulty fonts rather than handling things tidily.